sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -137,12 +137,13 @@ case class InsertIntoTable(
   }
 }
 
-case class CreateTableAsSelect[T](
-    databaseName: Option[String],
-    tableName: String,
-    child: LogicalPlan,
-    allowExisting: Boolean,
-    desc: Option[T] = None) extends UnaryNode {
+trait CreateTableAsSelect extends UnaryNode {
+  self: Product =>
+  def databaseName: Option[String]
+  def tableName: String
+  def child: LogicalPlan
+  def allowExisting: Boolean
+
```

Comment:
I guess we cannot use `Command` because a `Command` is a `LeafNode`. If we make it a `Command`, the child logical plan will not be analyzed and optimized.